celltype_annotation/scripts/cell-assign.py

```diff
@@ -13,45 +13,68 @@
 from scvi.external import CellAssign
 import pandas as pd
 import numpy as np
+import argparse
+import re
 
-# read in public references as marker gene tables
-fl_file = os.path.join("..", "references", "FL_celltype_ensembl.csv")
-fl_mtx = pd.read_csv(fl_file, index_col=0)
+parser = argparse.ArgumentParser()
+parser.add_argument('-i', '--input_anndata',
+                    dest = 'input_anndata',
+                    required = True,
+                    help = 'Path to HDF5 file with processed AnnData object to annotate')
+parser.add_argument('-o', '--output_predictions',
+                    dest = 'output_predictions',
+                    required = True,
+                    help = 'Path to directory to save the predictions, should end in tsv')
+parser.add_argument('-r', '--reference',
+                    dest = 'reference',
+                    required = True,
+                    help = 'Path to marker by cell type reference file')
 
-fl_noBcells_file = os.path.join("..", "references", "FL_celltype_ensembl-noBcells.csv")
-fl_noBcells_mtx = pd.read_csv(fl_noBcells_file, index_col=0)
+args = parser.parse_args()
+
+# compile extension regex
+file_ext = re.compile(r"\.hdf5$|.h5$", re.IGNORECASE)
+
+# check that input file exists, if it does exist, make sure it's an h5 file
+if not os.path.exists(args.input_anndata):
+    raise FileExistsError("--input_anndata file not found.")
+elif not file_ext.search(args.input_anndata):
+    raise ValueError("--input_anndata must end in either .hdf5 or .h5 and contain a processed AnnData object.")
+
+# check that marker file exists and make sure its a csv
+if not os.path.exists(args.reference):
+    raise FileExistsError("--reference file not found.")
+elif not ".csv" in args.reference:
+    raise ValueError("--reference must be a csv file")
+
+# make sure output file path is tsv file
+if not ".tsv" in args.output_predictions:
+    raise ValueError("--output_predictions must provide a file path ending in tsv")
+
+# check that output file directory exists and create directory if doesn't exist
+predictions_dir = os.path.dirname(args.output_predictions)
+os.makedirs(predictions_dir, exist_ok = True)
+
+# read in references as marker gene tables
+ref_matrix = pd.read_csv(args.reference, index_col=0)
 
 # file path to annotated sce
-anndata_dir = os.path.join("..", "data", "anndata")
-annotated_adata_file = os.path.join(anndata_dir, "SCPCL000295_anndata.hdf5")
-annotated_adata = adata.read_h5ad(annotated_adata_file)
+annotated_adata = adata.read_h5ad(args.input_anndata)
 
 # index adata but use row data to do it
-subset_adata = annotated_adata[:, fl_mtx.index].copy()
+subset_adata = annotated_adata[:, ref_matrix.index].copy()
 subset_adata.X = subset_adata.X.tocsr()
```

Comment:
This unfortunately is required for scvi tools. Otherwise you get errors about not being the correct format. I followed the same code we use in https://github.com/AlexsLemonade/sc-data-integration/blob/3994b1934e255a129d5efcb56fa5fa2130d075dc/scripts/utils/integrate_scvi.py#L104-L108